src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -141,15 +146,42 @@ class WorkspaceService(userInfo: UserInfo, dataSource: DataSource, workspaceDAO:
         case None =>
           val bucketName = createBucketName(workspaceRequest.name)
           Try( gcsDAO.createBucket(userInfo.userId,workspaceRequest.namespace, bucketName) ) match {
-            case Failure(err) => RequestComplete(StatusCodes.Forbidden, s"Unable to create bucket for ${workspaceRequest.namespace}/${workspaceRequest.name}: " + err.getMessage)
+            case Failure(err) =>
+              RequestComplete(StatusCodes.Forbidden,s"Unable to create bucket for ${workspaceRequest.namespace}/${workspaceRequest.name}: "+err.getMessage)
             case Success(_) =>
-              val workspace = Workspace(workspaceRequest.namespace, workspaceRequest.name, bucketName, DateTime.now, userInfo.userId, workspaceRequest.attributes)
-              workspaceDAO.save(workspace, txn)
-              PerRequest.RequestComplete((StatusCodes.Created, workspace))
+              Try( setupWorkspaceGroupACLs(WorkspaceName(workspaceRequest.namespace, workspaceRequest.name), bucketName) ) match {
+                case Failure(err) =>
+                  dataSource.rollbackOnly.set(true)
```

Comment:
if setupWorkspaceGroupACLs doesnt modify the graph it wouldn't be needed, nor would inTransaction - we would instead need some sort of google group rollback mechanism in setupWorkspaceGroupACLs if we had a problem in any line in there and wanted to get back to previous state